### PR TITLE
Add a script to change file names when upstreaming to OpenZFS/illumos

### PR DESCRIPTION
**Pull request by ahrens**

### CODE REVIEW

File: scripts/zol2zfs-patch.sed [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+s:include/libzfs.h:usr/src/lib/libzfs/common/libzfs.h:g
+s:man/man8/zfs.8:usr/src/man/man1m/zfs.1m:g
+s:include/sys:usr/src/uts/common/sys:g
+s:include/libzfs_core.h:usr/src/lib/libzfs_core/common/libzfs_core.h:g
+s:include/zfs_fletcher.h:usr/src/common/zfs/zfs_fletcher.h:g
+s:module/zcommon:usr/src/common/zfs/:g
+s:lib/libzfs:usr/src/lib/libzfs/common:g
+s:lib/libzfs_core:usr/src/lib/libzfs_core/common:g
+s:include/sys:lib/libzpool/common/sys:g
+s:lib/libzpool:lib/libzpool/common:g
```
Comment (prashks) [inline scripts/zol2zfs-patch.sed]: Also need this for the zfs-tests -

s:tests/zfs-tests:test/zfs-tests:g

File: scripts/zol2zfs-patch.sed [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sed -f
```
Comment (prashks) [inline scripts/zol2zfs-patch.sed]: Would be nice to determine the sed path automatically. 
For example, I had to change this to _/usr/bin/sed_ when running on my mac.

File: scripts/zol2zfs-patch.sed (hunk 1 above)
Comment (ahrens) [inline scripts/zol2zfs-patch.sed]: done, thanks

Comment (prashks) [inline scripts/zol2zfs-patch.sed]: Couple more -
`s:include:usr/src/common/zfs:g
  s:man/man5/zpool-features.5:usr/src/man/man5/zpool-features.5:g`

Comment (ahrens) [inline scripts/zol2zfs-patch.sed]: done!

**Pull request merged**